pkg/build/api/validation/validation.go

```diff
@@ -36,22 +36,43 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) fielderro
 	return allErrs
 }
 
+func streamTagKey(ref *kapi.ObjectReference) string {
+	if ref.Namespace != "" {
+		return fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+	}
+	return ref.Name
```

Comment:
shouldn't we prepend the namespace of the object containing the ref, by default?  (ie if the ref doesn't indicate a namespace, the namespace should be the namespace the ref lives in)